cmd/nitro/nitro.go

```diff
@@ -564,6 +565,27 @@ func (c *NodeConfig) Validate() error {
 	return c.Node.Validate()
 }
 
+type RpcLogger struct{}
+
+func (l RpcLogger) OnRequest(request interface{}) rpc.ResultHook {
+	log.Trace("sending L1 RPC request", "request", request)
+	return RpcResultLogger{request}
+}
+
+type RpcResultLogger struct {
+	request interface{}
+}
+
+func (l RpcResultLogger) OnResult(response interface{}, err error) {
+	if err != nil {
+		// The request might not've been logged if the log level is debug not trace, so we log it again here
+		log.Debug("received error response from L1 RPC", "request", l.request, "response", response, "err", err)
```

Comment:
A warning or at least info seems in place?

Reply:
I'm just not sure under what conditions this might appear, and don't want to concern users unnecessarily. But I'll make it an info :+1: